events/code_review_events/workflow.py

```diff
@@ -299,73 +300,114 @@ def __init__(self, cache_root):
         # Create message bus shared amongst processes
         self.bus = MessageBus()
 
-        self.workflow = CodeReview(
-            api_key=taskcluster_config.secrets["PHABRICATOR"]["api_key"],
-            url=taskcluster_config.secrets["PHABRICATOR"]["url"],
-            publish=taskcluster_config.secrets["PHABRICATOR"].get("publish", False),
-            risk_analysis_reviewers=taskcluster_config.secrets.get(
-                "risk_analysis_reviewers", []
-            ),
-            community_config=taskcluster_config.secrets.get("taskcluster_community"),
-            user_blacklist=taskcluster_config.secrets["user_blacklist"],
-        )
-        self.workflow.register(self.bus)
-
-        # Build mercurial worker and queue
-        self.mercurial = MercurialWorker(
-            QUEUE_MERCURIAL,
-            QUEUE_PHABRICATOR_RESULTS,
-            repositories=self.workflow.get_repositories(
-                taskcluster_config.secrets["repositories"], cache_root
-            ),
-        )
-        self.mercurial.register(self.bus)
+        publish = taskcluster_config.secrets["PHABRICATOR"].get("publish", False)
 
-        # Create web server
-        self.webserver = WebServer(QUEUE_WEB_BUILDS)
-        self.webserver.register(self.bus)
+        # Check the redis support is enabled on Heroku
+        if heroku.in_dyno():
+            assert self.bus.redis_enabled is True, "Need Redis on Heroku"
 
-        # Setup monitoring for newly created tasks
-        self.monitoring = Monitoring(
-            QUEUE_MONITORING, taskcluster_config.secrets["admins"], MONITORING_PERIOD
-        )
-        self.monitoring.register(self.bus)
+        # Run webserver & pulse on web dyno or single instance
+        if not heroku.in_dyno() or heroku.in_web_dyno():
+
+            # Create web server
+            self.webserver = WebServer(QUEUE_WEB_BUILDS)
+            self.webserver.register(self.bus)
 
-        # Create pulse listener for unit test failures
-        if self.workflow.publish:
+            # Create pulse listener for unit test failures
             self.pulse = PulseListener(
```

Comment:
Pulse connection should be made only when publish is True (as we are only consuming the queue when publish is True, otherwise we will always overgrow the queue).

Reply:
ok, but i have to keep the webserver process running

Reply:
Sure, it's just about the pulse connection.

Reply:
(to keep the same behavior we had before this PR)